### DISCUSSION TITLE
Compile without C++11

### DISCUSSION THREAD

**Discussion by aggelosmots** [nasa/fprime#2681] [Q&A]
Hello,

In the latest fprime version, there is a requirement that compiler should support C++11 features. But in several discussions, creators mention that the development of fprime started with C++98, so I assumed that the framework should compile with std=c++0x flag. I did a try to compile fprime with GCC 4.4 version, but no luck I had compile errors. So my questions are:
* Is there a way to compile fprime using compilers with no c++11? If yes what are the steps?
* If not is the C++98 supported until a specific version? If yes which one?
* If no to everything is there any other approach to use fprime without C++11?

Thank you

**Comment by LeStarch** [comment]
fprime officially dropped support for C++98 with version v3.0.0. fprime versions older than that should still support c++98.

GCC 4 is quite old.  Would it be possible to elaborate a bit on the use case here?  We'd like to understand your use case a bit more.

**Reply by aggelosmots** [reply]
I'm working with an AVR32 which has GCC 4.4.7. I can't find newer compilers, so the latest C++ features I can use are those before the C++11, just those of C++0X which did not work on v3.4.3. I just had done an attempt with fprime v2.1.0 and I had some issues with the dependencies, but I managed to compile a simple project w/out a deployment. I haven't tested it yet with the avr32 compiler.
